src/main/java/org/opensearch/knn/index/KNNIndexShard.java

```diff
@@ -88,16 +93,27 @@ public String getIndexName() {
     }
 
     /**
-     * Load all of the HNSW graphs for this shard into the cache. Note that getIndices is called to prevent loading
-     * in duplicates.
+     * Load all of the k-NN segments for this shard into the cache.
      *
      * @throws IOException Thrown when getting the HNSW Paths to be loaded in
      */
     public void warmup() throws IOException {
         logger.info("[KNN] Warming up index: " + getIndexName());
         try (Engine.Searcher searcher = indexShard.acquireSearcher("knn-warmup")) {
-            Map<String, SpaceType> allEnginePaths = getAllEnginePaths(searcher.getIndexReader());
-            knnIndexCache.loadIndices(allEnginePaths, getIndexName());
+            getAllEnginePaths(searcher.getIndexReader()).entrySet().stream().map(e ->
```

Comment:
why not merge it into single map/foreach?

Reply:
Yeah I can refactor.